x-pack/legacy/plugins/watcher/public/np_ready/application/app.tsx

```diff
@@ -4,54 +4,63 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
-import { HashRouter, Switch, Route, Redirect } from 'react-router-dom';
+import React from 'react';
+import {
+  ChromeStart,
+  DocLinksStart,
+  HttpSetup,
+  SavedObjectsClientContract,
+  ToastsSetup,
+  UiSettingsClientContract,
+} from 'src/core/public';
+import { EuiCallOut, EuiLink } from '@elastic/eui';
+import {
+  HashRouter,
+  Switch,
+  Route,
+  Redirect,
+  withRouter,
+  RouteComponentProps,
+} from 'react-router-dom';
 import { FormattedMessage } from '@kbn/i18n/react';
 import { WatchStatus } from './sections/watch_status/components/watch_status';
 import { WatchEdit } from './sections/watch_edit/components/watch_edit';
 import { WatchList } from './sections/watch_list/components/watch_list';
 import { registerRouter } from './lib/navigation';
 import { BASE_PATH } from './constants';
-import { LICENSE_STATUS_VALID } from '../../../common/constants';
-import { EuiCallOut, EuiLink } from '@elastic/eui';
-
-class ShareRouter extends Component {
-  static contextTypes = {
-    router: PropTypes.shape({
-      history: PropTypes.shape({
-        push: PropTypes.func.isRequired,
-        createHref: PropTypes.func.isRequired
-      }).isRequired
-    }).isRequired
-  }
-  constructor(...args) {
-    super(...args);
-    this.registerRouter();
-  }
+import { LICENSE_STATUS_VALID } from '../../../../../common/constants';
+import { AppContextProvider } from './app_context';
 
-  registerRouter() {
-    // Share the router with the app without requiring React or context.
-    const { router } = this.context;
-    registerRouter(router);
-  }
+const ShareRouter = withRouter(({ children, history }: RouteComponentProps & { children: any }) => {
+  registerRouter({ history });
+  return children;
+});
 
-  render() {
-    return this.props.children;
-  }
+export interface AppDeps {
```

Comment:
Similar to `ServerShimWithRouter`, `AppDeps` and `BootDeps` make it hard to tell apart legacy from NP.